_tests_/components/MainButton.test.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { MouseEventHandler } from 'react';
+
+import { MainButton } from '@components/common/buttons/mainButtons';
+import { MainButtonProps } from '@components/common/buttons/types/mainButtons';
+
+import { render, screen, fireEvent } from '@testing-library/react';
+
+describe('MainButton', () => {
+  const defaultProps: MainButtonProps = {
+    text: '메인버튼',
+    variant: 'primary',
+    icon: undefined,
+    disabled: false,
+    onClick: jest.fn(),
+    type: 'button',
+  };
+
+  beforeEach(() => {
+    (defaultProps.onClick as jest.MockedFunction<MouseEventHandler<HTMLButtonElement>>).mockClear();
+  });
+
+  it('text값을 가진 버튼이 렌더링 된다.', () => {
+    render(<MainButton {...defaultProps} />);
+    expect(screen.getByText('메인버튼')).toBeInTheDocument();
+  });
+
+  it('React.Element타입의 아이콘이 버튼에 렌더링된다.', () => {
+    render(<MainButton {...defaultProps} icon={<span data-testid='icon'>👍</span>} />);
+    expect(screen.getByTestId('icon')).toBeInTheDocument();
```

Comment:
요기서 실제로 icon을 적용하는 방식으로 확인해보는 것도 좋을 것 같다고 생각했어요!

```suggestion
    render(
      <MainButton
        {...defaultProps}
        icon={<Image src={IconPencil} alt='연필 아이콘' data-testid='icon' />}
      />,
    );
    expect(screen.getByTestId('icon')).toBeInTheDocument();
    expect(screen.getByAltText('연필 아이콘'));
```

Reply:
https://github.com/dreamyPatisiel/devdevdev-client/pull/127/commits/9b14025ac02e75880dccd29163d3a405e6c1fbf8 😊😊

Reply:
alt도 확인하면 어떤가욤??!

Reply:
[3338c84](https://github.com/dreamyPatisiel/devdevdev-client/pull/127/commits/3338c84fb937517e9eeacc4c85dfe35745e5d713) 😊😊